### DISCUSSION TITLE
ptxas version issue

### DISCUSSION THREAD

**Discussion by zohimchandani** [jax-ml/jax#10327] [Q&A]
Getting the following warning message but not sure how to install `ptxas`? Are there instructions on how to do this somewhere?

```

2022-04-16 11:15:22.281636: W external/org_tensorflow/tensorflow/stream_executor/gpu/asm_compiler.cc:111] *** WARNING *** You are using ptxas 10.0.145, which is older than 11.1. ptxas before 11.1 is known to miscompile XLA code, leading to incorrect results or invalid-address errors.

You may not need to update to CUDA 11.1; cherry-picking the ptxas binary is often sufficient.

```

**Comment by YouJiacheng** [comment]
`ptxas` is a part of CUDA toolkit.
I suggest you to use `conda` to install CUDA.(Install CUDA by yourself can easily waste 1 day)
```
conda install cuda -c nvidia
```

**Reply by rtdew1** [reply]
I realize this is old, but thanks for this! Saved me a lot of time on a similar error...

**Reply by BwandoWando** [reply]
> conda install cuda -c nvidia

amazing, thank you. It works.

**Reply by Yang-Xijie** [reply]
Yeah! You save my life! Thanks a lot!
Some supplements:
- If you dont want to install the latest version, you can check https://anaconda.org/nvidia/repo/installers and choose a lable with the desired version. And change the "package" to "cuda".
  - eg. I want to install cuda 11.8 so I use the command `conda install --channel "nvidia/label/cuda-11.8.0" cuda`

**Comment by binpan0423** [comment]
> Getting the following warning message but not sure how to install `ptxas`? Are there instructions on how to do this somewhere?
> 
> ```
> 
> 2022-04-16 11:15:22.281636: W external/org_tensorflow/tensorflow/stream_executor/gpu/asm_compiler.cc:111] *** WARNING *** You are using ptxas 10.0.145, which is older than 11.1. ptxas before 11.1 is known to miscompile XLA code, leading to incorrect results or invalid-address errors.
> 
> You may not need to update to CUDA 11.1; cherry-picking the ptxas binary is often sufficient.
> ```

did you solve this problem？

**Reply by hawkinsp** [reply]
This means that the version of CUDA in your `PATH` is too old. Try running `which ptxas` to determine which copy of CUDA is being used.

JAX requires CUDA 11.1 or newer: https://github.com/google/jax#pip-installation-gpu-cuda

Hope that helps!

**Comment by 1evergreen** [comment]
If there are many cuda toolkits of different version , manually set $CUDA in the .bashrc file and set $CUDA/bin to the $PATH enviroment. then use which ptxas to check if it is the right version.

If you are using conda env, set to the activate config file instead

**Reply by conorheins** [reply]
sorry for my naivete, but can you explain how to do the following:
```
manually set $CUDA in the .bashrc file and set $CUDA/bin to the $PATH enviroment.
```
